security/rbac/engine/engine.go

```diff
@@ -84,49 +85,49 @@ func exprToProgram(condition *expr.Expr) *cel.Program {
 // Evaluates a CEL expression. Returns true if CEL evaluation returns true. Returns false if
 // CEL evaluation returns false. Returns an error if CEL evaluation returns unknown or an
 // unsuccessful evaluation.
-func evaluateCondition(condition *cel.Program, evalMap map[string]interface{}) (bool, error) {
+func evaluateCondition(condition *cel.Program, activation interpreter.Activation) (bool, error) {
 	// TODO(@ezou): implement the matching logic using CEL library.
 	return false, nil
 }
 
-// celEngine is the struct for an engine created from one RBAC proto.
-type celEngine struct {
+// policyEngine is the struct for an engine created from one RBAC proto.
+type policyEngine struct {
 	action     pb.RBAC_Action
 	conditions map[string]*cel.Program
 }
 
-// Creates a new celEngine from an RBAC policy proto.
-func newCelEngine(rbac pb.RBAC) celEngine {
+// Creates a new policyEngine from an RBAC policy proto.
+func newPolicyEngine(rbac pb.RBAC) policyEngine {
 	action := rbac.Action
 	conditions := make(map[string]*cel.Program)
 	for policyName, policy := range rbac.Policies {
 		conditions[policyName] = exprToProgram(policy.Condition)
 	}
-	return celEngine{action, conditions}
+	return policyEngine{action, conditions}
 }
 
 // Returns the decision of an engine based on whether or not AuthorizationArgs is a match,
 // i.e. if engine's action is ALLOW and match is true, we will return DecisionAllow;
 // if engine's action is ALLOW and match is false, we will return DecisionDeny.
-func getDecision(engine celEngine, match bool) Decision {
+func getDecision(engine policyEngine, match bool) Decision {
 	if engine.action == pb.RBAC_ALLOW && match || engine.action == pb.RBAC_DENY && !match {
 		return DecisionAllow
 	}
 	return DecisionDeny
 }
 
-// Returns the authorization decision of a single engine based on evalMap.
+// Returns the authorization decision of a single policy engine based on activation.
 // If any policy matches, the decision matches the engine's action, and the list of
 //  matching policy names will be returned.
 // Else if any policy is missing attributes, the decision is unknown, and the list of
 //  policy names that can't be evaluated due to missing attributes will be returned.
 // Else, the decision is the opposite of the engine's action, i.e. an ALLOW engine
 //  will return DecisionDeny, and vice versa.
-func (engine celEngine) evaluate(evalMap map[string]interface{}) (Decision, []string) {
+func (engine policyEngine) evaluate(activation interpreter.Activation) (Decision, []string) {
```

Comment:
If we plan to stop evaluating after a successful match, then logic could be simpler and cleaner.
Could you please add a todo before line 126.

```
TODO: Revisit the implementation if we plan to stop evaluation after a successful match.
```